examples/job/expansions/README.md

```diff
@@ -0,0 +1,237 @@
+```
+
+[Download example](job-template.yaml?raw=true)
+<!-- END MUNGE: EXAMPLE job-template.yaml -->
+
+Unlike a *pod template*, a job template is not a Kubernetes API type.  It is just
+a Job object that has some placeholders that need to be filled in before it
+can be used.
+
+In this example, the only processing the container does is to `echo` a string and
+sleep for a bit.
+
+In a real use case, the processing would be some substantial computation, such as
+rendering a frame of a movie, or processing a range of rows in a database.
+The "$ITEM" parameter would specify for example, the frame number or the row range.
+
+This Job has two labels. The first label, with key `app`, distinguishes this group of jobs from
+other groups of jobs (these are not shown, but there might be other ones).  The second key (`item`)
+distinguishes individual jobs in the group.  Neither of these keys are special to kubernetes --
+you can pick your own label scheme.  Just make sure each job has a different label selector, from
+every other job.
+
+Next, expand the template into multiple files, one for each item to be processed.
+
+```console
+# Expand files into a temporary directory
+$ mkdir ./tmp
+$ for i in apple banana cherry
+do
+  cat job-template.yaml | sed "s/\$ITEM/$i/" > ./tmp/job-$i.yaml
+done
+$ ls jobs/
+job-apple.yaml
+job-banana.yaml
+job-cherry.yaml
+```
+
+Here, we used sed to fill in the template, but you could use any type of template
+language, such as jinja2, erb, etc, or generate the items from a script.
+
+Now create all the jobs with one kubectl command:
+
+```console
+$ kubectl create -f ./jobs
+job "jobexample-apple" created
+job "jobexample-banana" created
+job "jobexample-cherry" created
+```
+
+Now, check on the jobs:
+
+```console
+$ kubectl get jobs -l app=jobexample
+JOB                CONTAINER(S)   IMAGE(S)   SELECTOR                               SUCCESSFUL
+jobexample-apple   c              busybox    app in (jobexample),item in (apple)    1
+jobexample-banana  c              busybox    app in (jobexample),item in (banana)   0
+jobexample-cherry  c              busybox    app in (jobexample),item in (cherry)   0
+```
+
+Here we use the `-l` option to select all jobs that are part of this
+group of jobs.  (There might be other unrelated jobs in the system that we
+do not care to see.)
+
+We can check on the pods as well using the same label selector:
+
+```console
+# Now check on their pods
+$ kubectl get pods -l app=jobexample
+NAME                      READY     STATUS      RESTARTS   AGE
+jobexample-apple-43cz3    0/1       Completed   0          1m
+jobexample-banana-bv4s8   0/1       Completed   0          1m
+jobexample-cherry-ml2rv   0/1       Completed   0          1m
+```
+
+There is not a single command to check on the output of all jobs at once,
+but looping over all the pods is pretty easy:
+
+```console
+# Get logs of all the pods.
+$ for p in $(kubectl get pods -l app=jobexample -o name | cut -f 2 -d "/")
```

Comment:
Even more you're using that pattern in work-queue-2, towards the end of README.md.